akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/EventsourcedRunning.scala

```diff
@@ -3,39 +3,40 @@
  */
 package akka.persistence.typed.internal
 
-import akka.actor.typed.Behavior
+import akka.actor.typed.{ Behavior, Signal }
 import akka.actor.typed.Behavior.StoppedBehavior
+import akka.actor.typed.scaladsl.Behaviors
 import akka.actor.typed.scaladsl.Behaviors.MutableBehavior
-import akka.actor.typed.scaladsl.{ ActorContext, Behaviors, StashBuffer, TimerScheduler }
 import akka.annotation.InternalApi
-import akka.event.Logging
 import akka.persistence.Eventsourced.{ PendingHandlerInvocation, StashingHandlerInvocation }
 import akka.persistence.JournalProtocol._
 import akka.persistence._
 import akka.persistence.journal.Tagged
-import akka.persistence.typed.internal.EventsourcedBehavior.InternalProtocol
-import akka.persistence.typed.internal.EventsourcedBehavior.InternalProtocol.{ IncomingCommand, JournalResponse, RecoveryTickEvent, SnapshotterResponse }
+import akka.persistence.typed.internal.EventsourcedBehavior.{ InternalProtocol, MDC }
+import akka.persistence.typed.internal.EventsourcedBehavior.InternalProtocol.{ IncomingCommand, JournalResponse, SnapshotterResponse }
 
 import scala.annotation.tailrec
 import scala.collection.immutable
 
 /**
  * INTERNAL API
  *
- * Fourth (of four) -- also known as 'final' or 'ultimate' -- form of PersistentBehavior.
+ * Conceptually fourth (of four) -- also known as 'final' or 'ultimate' -- form of PersistentBehavior.
  *
  * In this phase recovery has completed successfully and we continue handling incoming commands,
  * as well as persisting new events as dictated by the user handlers.
  *
- * This behavior operates in two phases:
+ * This behavior operates in two phases (also behaviors):
  * - HandlingCommands - where the command handler is invoked for incoming commands
  * - PersistingEvents - where incoming commands are stashed until persistence completes
  *
  * This is implemented as such to avoid creating many EventsourcedRunning instances,
  * which perform the Persistence extension lookup on creation and similar things (config lookup)
  *
+ * See previous [[EventsourcedRecoveringEvents]].
  */
-@InternalApi private[akka] object EventsourcedRunning {
+@InternalApi
+private[akka] object EventsourcedRunning {
```

Comment:
the simplifications of `pendingInvocations` are not done, I can take a stab at that if you like

Reply:
Here is the simplification of the pendingInvocations: fc5e5cfbd1079c1ffec7d57004c8c5496b3d7758
Probably best to apply that before other changes.